community/kernel/src/test/java/org/neo4j/kernel/impl/api/OperationsFacadeTest.java

```diff
@@ -1,135 +0,0 @@
-import org.neo4j.kernel.api.schema.SchemaDescriptorFactory;
-import org.neo4j.kernel.api.schema.index.SchemaIndexDescriptor;
-import org.neo4j.kernel.api.schema.index.SchemaIndexDescriptorFactory;
-import org.neo4j.test.rule.DatabaseRule;
-import org.neo4j.test.rule.ImpermanentDatabaseRule;
-
-import static org.hamcrest.CoreMatchers.allOf;
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.Matchers.containsString;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.fail;
-
-public class OperationsFacadeTest
-{
-    @Rule
-    public DatabaseRule db = new ImpermanentDatabaseRule();
-
-    private final Label LABEL1 = Label.label( "Label1" );
-    private final String PROP1 = "Prop1";
-    private int labelId;
-    private int propertyId;
-
-    @Before
-    public void setup()
-    {
-        try ( Transaction tx = db.beginTx() )
-        {
-            db.createNode( LABEL1 ).setProperty( PROP1, 1 );
-            tx.success();
-        }
-
-        try ( Transaction tx = db.beginTx();
-              Statement statement = db.statement() )
-        {
-            ReadOperations readOperations = statement.readOperations();
-            labelId = readOperations.labelGetForName( LABEL1.name() );
-            propertyId = readOperations.propertyKeyGetForName( PROP1 );
-            tx.success();
-        }
-    }
-
-    @Test
-    public void testThrowExceptionWhenIndexNotFound()
-    {
-        try ( Transaction ignored = db.beginTx();
-              Statement statement = db.statement() )
-        {
-            ReadOperations readOperations = statement.readOperations();
-            try
-            {
-                LabelSchemaDescriptor descriptor = SchemaDescriptorFactory.forLabel( 1, 2 );
-                readOperations.indexGetForSchema( descriptor );
-                fail( "Should have failed" );
-            }
-            catch ( SchemaRuleNotFoundException e )
-            {
-                assertThat( e.getMessage(), containsString( "No index was found for :label[1](property[2])" ) );
-            }
-        }
-    }
-
-    @Test
-    public void indexGetProviderDescriptorMustReturnUndecidedIfIndexCreatedInTransaction() throws Exception
-    {
-        try ( Transaction tx = db.beginTx();
-              Statement statement = db.statement() )
-        {
-            db.schema().indexFor( LABEL1 ).on( PROP1 ).create();
-            ReadOperations readOperations = statement.readOperations();
-            SchemaIndexDescriptor schemaIndexDescriptor = SchemaIndexDescriptorFactory.forLabel( labelId, propertyId );
-            IndexProvider.Descriptor providerDescriptor = readOperations.indexGetProviderDescriptor(
-                    schemaIndexDescriptor );
-            assertThat( providerDescriptor, is( IndexProvider.UNDECIDED ) );
-            tx.success();
-        }
-    }
-
-    @Test
-    public void indexGetProviderDescriptorMustThrowIfIndexDoesNotExist()
```

Comment:
An equivalence of this one too?

Reply:
SchemaRead.index will no longer throw, it will return empty index. That behaviour is tested.